File: server/MotoRevApi/Model/Concessionaria.cs
namespace MotoRevApi.Model;

public class Concessionaria
{
    public int Id { get; set; }
    public required string Nome { get; set; }
    public required string Cnpj { get; set; }
    public required string Telefone { get; set; }
    public string Tipo { get; set; } = "Matriz";
    public required string UsuarioId { get; set; }
    public virtual Usuario Usuario { get; set; } = null!;
    public virtual ICollection<Loja> Lojas { get; set; } = [];
}

File: client/src/app/models/Concessionaria.ts
import { Loja } from '@/app/models/Loja';

export interface Concessionaria {
  id: number;
  nome: string;
  email: string;
  cnpj: string;
  telefone: string;
  tipo: string;
  cep: string;
  logradouro: string;
  numero: string;
  bairro: string;
  cidade: string;
  uf: string;
  lojas: Loja[];
}
